README.md

```diff
@@ -299,3 +298,39 @@ NOTE: GenQuery2 column names are subject to change as the implementation is impr
 | TICKET_ALLOWED_USER_TICKET_ID | R_TICKET_ALLOWED_USERS | ticket_id |
 | TICKET_ALLOWED_GROUP_NAME | R_TICKET_ALLOWED_GROUPS | group_name |
 | TICKET_ALLOWED_GROUP_TICKET_ID | R_TICKET_ALLOWED_GROUPS | ticket_id |
+
+## Questions and Answers
+
+### Can I use the GenQuery2 microservices with the Python Rule Engine Plugin (PREP)?
+
+Yes! The important thing to remember is that the GenQuery2 microservices can update input arguments. This behavior is fully supported by the iRODS Rule Language, but not the PREP. Changes to input arguments are only visible via the object returned from the microservice.
+
+Below is an example demonstrating how to use the GenQuery2 microservices with the PREP. Pay close attention to the objects returned from each microservice call.
+
+```python
+def list_all_data_objects(rule_args, callback, rei):
+    # Notice the first input argument. In the iRODS Rule Language, the microservice
+    # would write the context handle to it. Here, it is only a placeholder.
+    res = callback.genquery2_execute('', 'select COLL_NAME, DATA_NAME')
+
+    # We can retrieve the context handle by reading the "arguments" list like so. The
+    # "arguments" list is zero-indexed and each index maps to an argument in the call
+    # previously made.
+    ctx = res['arguments'][0]
```

Comment:
consider 'context' and 'result' in the example...   but everything else seems good and correct.

Reply:
Sure thing.

Reply:
Changed those everywhere for consistency.